### PR TITLE
reworked Deprecated tags test

### PR DESCRIPTION
**Pull request by RomanFedytskyi**

### CODE REVIEW

File: tests/testtags.js [hunk 1]
```diff
@@ -1,46 +1,89 @@
 /**
- * @fileoverview Defines tests for deprecated tags checker.
+ * @fileoverview Defines test for Deprecated tags widget.
+ * - Element with CSS class 'widget-deprecated-tags' should exist;
```
Comment (mshchygol) [inline tests/testtags.js:4]: Please add, that this are 'Success criterias '

File: tests/testtags.js [hunk 2]
```diff
@@ -1,46 +1,90 @@
...
- * @fileoverview Defines tests for deprecated tags checker.
+ * @fileoverview Defines test for Deprecated tags widget.
+ * Success criteria:
+ * - Element with CSS class 'widget-deprecated-tags' should exist;
+ * - Element with CSS class 'widget-deprecated-tags' should contain element
+ *   with ID 'content-tags-container';
+ * - Element with ID 'content-tags-container' should contain element with ID
+ *   'tags-data-table';
+ * - If element with CSS class 'widget-deprecated-tags' isn't empty, element
+ *   with ID 'tags-data-table' shuld contain elements with tag name 'tr';
```
Comment (mshchygol) [inline tests/testtags.js]: ...should

Comment (RomanFedytskyi) [inline tests/testtags.js]: fixed

File: tests/testtags.js [hunk 3]
```diff
@@ -1,46 +1,90 @@
...
+      for (; length;) {
+        result = !result;
+        // If element with CSS class 'widget-deprecated-tags' isn't empty,
+        // elements with tag name 'tr' should contain elements with 
+        // tag name 'th' and 'td'.
+        cells = nodes[--length].querySelectorAll('th, td');
+        number = cells[1].textContent.trim();
+        // In element with tag name 'tr' first element should contain string,
+        // second element should contain number.
+        if (!(cells[0].textContent.trim(  ) && number && PATTERN.test(number))) {
```
Comment (mshchygol) [inline tests/testtags.js]: I think, you don't need extra spaces here.

Comment (FeditskiyRoman) [inline tests/testtags.js]: fixed

**Comment by mshchygol** [comment]
LGTM

**Comment by AlbaMontis** [comment]
LGTM

**Comment by vasylsemiliak** [comment]
LGTM

**Comment by Alexsv7** [comment]
LGTM

File: tests/testtags.js [hunk 4]
```diff
@@ -1,46 +1,90 @@
...
+  // Element with CSS class 'widget-deprecated-tags' should contain element
+  //  with ID 'content-tags-container'.
   /** @type {Element} */
-  var trueElement = container && container.getElementsByClassName('true')[0];
+  var container = document.getElementById(CONTAINER_ID);
+
+  // If element with CSS class 'widget-deprecated-tags' is empty, it should
+  // contain element with CSS classes 'rule not true';
+  /** @type {boolean} */
+  var result = !container.querySelector(TRUE_SELECTOR);
```
Comment (vpodk) [inline tests/testtags.js]: If `container` is `null`?

File: tests/testtags.js [hunk 5]
```diff
@@ -1,46 +1,90 @@
...
   /** @type {Element} */
-  var passElement = container && container.getElementsByClassName('pass')[0];
-  /** @type {string} */
-  var message = passElement && passElement.textContent.trim();
-  /** @type {Element} */ var table = document.getElementById('tags-data-table');
+  var passElement = container.querySelector(PASS_SELECTOR);
```
Comment (vpodk) [inline tests/testtags.js]: Same as above.

**Pull request merged**